### ISSUE TITLE
create hash for each mesh as a unique ID

### ISSUE THREAD

**Issue by jasonfleming** [StormSurgeLive/asgs#765]
If a mesh is modified in a subtle way (e.g. bathymetry change on a small number of nodes), it may be difficult to detect. Each mesh version should have a unique name and version number as well as a 256 bit hash. The hash should be stored in the run.properties file as well as netCDF global attributes to indicate the specific mesh data in the file.

**Comment by wwlwpd** [comment]
This is a good idea, I am wondering if there is a way to normalize the actual content in a repeatable way such that this is what is actually hashed rather than the "file". E.g., benign modifications (e.g., errant new lines at the beginning or end of the file) to the mesh should result in the same "hash" because it's based on the data and not any kind of accidental property of the file.

**Comment by jasonfleming** [comment]
Yes, hashing the file is the quick and dirty approach. To improve it, I wonder if we could use Nate's mesh reader from his ourPerl repo to read in the mesh, strip any leading/trailing white space or comments from each field, join it all into a single scalar string, and then hash that. Seems like the hash would only change then if the actual content changed.
